traceenv.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/otel/propagation"
+)
+
+const (
+	traceparentHeader = "traceparent"
+	tracestateHeader  = "tracestate"
+)
+
+// see https://github.com/moby/buildkit/pull/2572
+func initOtelContext(ctx context.Context) context.Context {
+	// open-telemetry/opentelemetry-specification#740
+	parent := os.Getenv("TRACEPARENT")
+	state := os.Getenv("TRACESTATE")
+
+	if parent != "" {
+		tc := propagation.TraceContext{}
+		return tc.Extract(ctx, &textMap{parent: parent, state: state})
+	}
+
+	// deprecated: removed in v0.11.0
```

Comment:
Do worth adding deprecated stuff?

Reply:
Extracted this entire block from Moby's buildkit. Maybe we can simply exclude that part, as we have a strong dependency on the opentelemetry SDK v1.1.0

Reply:
Removed, thanks!